### PR TITLE
Enhance appearance on artifacthub

### PR DESCRIPTION
**Pull request by NotMyFault**
Typo fixes are covered by https://github.com/jenkinsci/helm-charts/pull/843

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -3,9 +3,3 @@ contact_links:
   - name: Community forum
     url: https://community.jenkins.io/
     about: Please ask and answer questions here
-  - name:  Report a security vulnerability
-    url: https://github.com/jenkinsci/helm-charts/security/policy
-    about: Please review our security policy for more details
-  - name: Mailing lists
-    url: https://www.jenkins.io/mailing-lists/
-    about: You can also raise a question on the infra mailing list
```
Comment (NotMyFault) [inline .github/ISSUE_TEMPLATE/config.yml:11]: This is covered by the org template.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -100,4 +100,4 @@ Charts should start at `1.0.0`. Any breaking (backwards incompatible) changes to
 
 This project is released with a [Contributor Covenant](https://www.contributor-covenant.org).
 By participating in this project you agree to abide by its terms.
-See [CODE_OF_CONDUCT.md](./CODE_OF_CONDUCT.md).
+Visit our [code of conduct](https://www.jenkins.io/project/conduct/).
```
Comment (NotMyFault) [inline CONTRIBUTING.md:103]: Let's point to the maintained source of our CoC, rather to an unused file, covered by the org template.

File: charts/jenkins/Chart.yaml [hunk 3]
```diff
@@ -36,3 +40,5 @@ annotations:
       image: jenkins/inbound-agent:3107.v665000b_51092-5
     - name: backup
       image: maorfr/kube-tasks:0.2.0
+  artifacthub.io/category: "integration-delivery"
+  artifacthub.io/license: "Apache-2.0"
```
Comment (NotMyFault) [inline charts/jenkins/Chart.yaml:44]: Adds some additional information to our artifacthub info card. Thoughts on enabling the security graph? We do the same for docker images.

File: charts/jenkins/README.md [hunk 4]
```diff
@@ -46,20 +46,20 @@ $ helm upgrade [RELEASE_NAME] jenkins/jenkins [flags]
 
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
-Visit the chart's [CHANGELOG](./CHANGELOG.md) to view the chart's release history.
+Visit the chart's [CHANGELOG](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/CHANGELOG.md) to view the chart's release history.
```
Comment (NotMyFault) [inline charts/jenkins/README.md:49]: These links don't work on artifacthub, let's use the full URL.

File: charts/jenkins/Chart.yaml (hunk 3 above)
Comment (timja) [reply charts/jenkins/Chart.yaml:46]: sure

**Review by timja — APPROVED** [review]

File: charts/jenkins/Chart.yaml (hunk 3 above)
Comment (NotMyFault) [reply charts/jenkins/Chart.yaml:46]: Then someone just needs to enable it 😅

**Pull request merged**